src/mercury_engine_data_structures/formats/bmsbk.py

```diff
@@ -67,3 +67,17 @@ class Bmsbk(BaseResource):
     @functools.lru_cache
     def construct_class(cls, target_game: Game) -> Construct:
         return BMSBK
+
+    def get_block_group(self, block_group: int) -> Container:
+        return self.raw.block_groups[block_group]
+
+    def set_block_type(self, block_group: int, block_type: str) -> Container:
+        weakness = self.get_block_group(block_group).types[0]
```

Comment:
shouldn't it apply the new weakness to every block in the group? or am I misunderstanding